sky/engine/core/painting/OffsetBase.dart

```diff
@@ -4,19 +4,70 @@
 
 part of dart_ui;
 
+/// Base class for [Size] and [Offset], which are both ways to describe
+/// a distance as a two-dimensional axis-aligned vector.
 abstract class OffsetBase {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  ///
+  /// The first argument sets the horizontal dimension, and the second the
+  /// vertical dimension.
   const OffsetBase(this._dx, this._dy);
 
   final double _dx;
   final double _dy;
 
+  /// Returns true if either dimension is [double.INFINITY], and false if both
+  /// are finite (or negative infinity, or NaN).
+  ///
+  /// This is different than comparing for equality with an instance that has
+  /// _both_ dimensions set to [double.INFINITY].
   bool get isInfinite => _dx >= double.INFINITY || _dy >= double.INFINITY;
 
+  /// Less-than operator. Compares an [Offset] or [Size] to another [Offset] or
+  /// [Size], and returns true if both the horizontal and vertical values of the
+  /// left-hand-side operand are smaller than the horizontal and vertical values
+  /// of the right-hand-side operand respectively. Returns false otherwise.
+  ///
+  /// This is a partial ordering. It is possible for two values to be neither
+  /// less, nor greater than, nor equal to, another.
   bool operator <(OffsetBase other) => _dx < other._dx && _dy < other._dy;
```

Comment:
Unrelated to documentation, but I wonder if comparisons between Offset and Size should be disallowed via typing.